### PR TITLE
fix: invalid invocation error

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
fix illegal invocation error

### Screenshots [Optional]
image

## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/framework/src/hooks/useHtmlPassThrough.tsx [hunk 1]
```diff
@@ -23,14 +27,6 @@ export const useHtmlPassThrough = (
...
           });
         }
-
-        if (testId) {
-          try {
-            (node as HTMLElement).setAttribute("data-testid", testId);
-          } catch (e) {
-            error(e);
-          }
-        }
```
Comment (evshi) [inline packages/framework/src/hooks/useHtmlPassThrough.tsx:33]: Why does this fix the issue? It's the same call as what's in the for loop.

Comment (sagardspeed2) [reply packages/framework/src/hooks/useHtmlPassThrough.tsx:33]: @evshi honestly I have no idea that why this fix resolved by removing this, I just goes through the code structure and noticed that we no need to use this, as we can manage this with the for loop

Comment (anserwaseem) [reply packages/framework/src/hooks/useHtmlPassThrough.tsx:33]: Firefox shows the error more descriptively.
I believe this is the issue with the way we are using rootRef in antd Table. Getting the table containerzied inside a div with ref={rootRef} solves the problem without disturbing useHtmlPassThrough file

image

Comment (evshi) [reply packages/framework/src/hooks/useHtmlPassThrough.tsx:33]: Interesting - I wonder why putting it into a single map works. This probably means the ref changes...I think wrapping it in a div is reasonable.

**Review by evshi — COMMENTED** [review]
@sagardspeed2 you need to check the scroll behavior is not impacted by wrapping the div.

File: packages/framework/src/hooks/useHtmlPassThrough.tsx (hunk 1 above)
Comment (sagardspeed2) [reply packages/framework/src/hooks/useHtmlPassThrough.tsx:33]: @anserwaseem as per my view, no need to remove the changes from useHtmlPassThrough hook, as I just combine the 2 different attributes into one loop.

**Comment by sagardspeed2** [comment]
> @sagardspeed2 you need to check the scroll behavior is not impacted by wrapping the div.

@evshi yup, I tested scroll features on the datagrid and it is working after changing the refs to parent wrapper

**Review by evshi — APPROVED** [review]

File: packages/runtime/src/widgets/DataGrid/DataGrid.tsx [hunk 2]
```diff
@@ -277,7 +277,7 @@ export const DataGrid: React.FC<GridProps> = (props) => {
       const { scrollTop, scrollHeight, clientHeight } = container;
 
       // Check if the user has scrolled to the bottom
-      if (scrollTop + clientHeight === scrollHeight) {
+      if (Math.ceil(scrollTop + clientHeight) >= scrollHeight) {
```
Comment (anserwaseem) [inline packages/runtime/src/widgets/DataGrid/DataGrid.tsx:280]: please add this fix for onScrollEnd in GridView as well

**Review by anserwaseem — CHANGES_REQUESTED** [review]

File: packages/runtime/src/widgets/DataGrid/DataGrid.tsx (hunk 2 above)
Comment (sagardspeed2) [reply packages/runtime/src/widgets/DataGrid/DataGrid.tsx:280]: Added

**Review by anserwaseem — APPROVED** [review]

**Pull request merged**